GPU/Common/FramebufferManagerCommon.cpp

```diff
@@ -1660,99 +1656,52 @@ void FramebufferManagerCommon::FindTransferFramebuffers(VirtualFramebuffer *&dst
 		const u32 vfb_byteStride = vfb->fb_stride * vfb_bpp;
 		const u32 vfb_byteWidth = vfb->width * vfb_bpp;
 
-		// These heuristics are a bit annoying.
-		// The goal is to avoid using GPU block transfers for things that ought to be memory.
-		// Maybe we should even check for textures at these places instead?
-
-		if (vfb_address <= dstBasePtr && dstBasePtr < vfb_address + vfb_size) {
-			const u32 byteOffset = dstBasePtr - vfb_address;
-			const u32 byteStride = dstStride * bpp;
-			const u32 yOffset = byteOffset / byteStride;
+		if (vfb_address <= basePtr && basePtr < vfb_address + vfb_size) {
+			const u32 byteOffset = basePtr - vfb_address;
+			const u32 byteStride = stride * bpp;
+			const u32 memYOffset = byteOffset / byteStride;
 
-			// Some games use mismatching bitdepths.  But make sure the stride matches.
+			// Some games use mismatching bitdepths. But make sure the stride matches.
 			// If it doesn't, generally this means we detected the framebuffer with too large a height.
 			// Use bufferHeight in case of buffers that resize up and down often per frame (Valkyrie Profile.)
-			bool match = yOffset < dstYOffset && (int)yOffset <= (int)vfb->bufferHeight - dstHeight;
+
+			// TODO: Surely this first comparison should be <= ?
+			// Or does the exact match (byteOffset == 0) case get handled elsewhere?
```

Comment:
Added a new comment here, since I don't quite understand this.

Reply:
Quickly looking back at simpler times: a29d2e418e468173fe5c1c53f3367bb403e63381

Originally at least, this was just to make sure we picked the first one.  If two had the same y offset (including 0), only the first would be selected.

Looking back at modern times, `yOffset` is still unsigned and starting at 0xFFFFFFFF, which reasonable values of `memYOffset` are still less than, so it seems to perform the same function still.  That said, it might make more sense to evaluate multiple now, in case one has a CLUT usage...

-[Unknown]

Reply:
Ah I see, of course. However, I'm going to change this to use scoring with a sequence number bias, like framebuffer texturing. CLUT is not really a thing during block copies though.

Reply:
I'll do the rework in a separate commit though.

Reply:
I seem to remember Brave Story did a block transfer related to CLUTs.  This was basically a heuristic to allow the size that's otherwise abnormal - see 8e2557bafeba77c02d67deb3cb3aca27f439f581.

Not saying I would render to a framebuffer and then block transfer that to RAM, then specify that RAM as a CLUT.  But I didn't write Brave Story.

-[Unknown]